instant-xml/src/de.rs

```diff
@@ -108,17 +108,36 @@ impl<'xml> Iterator for Deserializer<'_, 'xml> {
             return None;
         }
 
-        let (prefix, local) = match self.context.next() {
-            Some(Ok(Node::Close { prefix, local })) => (prefix, local),
-            item => return item,
-        };
+        while let Some(Ok(node)) = self.context.next() {
+            match node {
+                Node::Close { prefix, local } => {
+                    let matches = self.context.stack.len() == self.level
+                        && local == self.local
+                        && prefix == self.prefix;
+
+                    // If the closing element doesn't match the current element, return an error.
+                    if !matches {
+                        return Some(Err(Error::UnexpectedState("close element mismatch")));
+                    }
 
-        if self.context.stack.len() == self.level && local == self.local && prefix == self.prefix {
-            self.done = true;
-            return None;
+                    // If the closing element matches the current element being deserialized,
+                    // mark the deserialization as done and return None to indicate the end of iteration.
+                    self.done = true;
+                    return None;
+                }
+                Node::Comment(_) => {
+                    // If the current node is a comment, skip it and continue to the next node.
+                    continue;
```

Comment:
For both of these other arms, maybe ditch the `;` from the values and move the comments to outside/before the arm instead of inside it, which should allow rustfmt to format this much more concisely. (Also I think the comments make more sense "on" the arm instead of "in" the arm.)